src/utils/gz.ts

```diff
@@ -2,107 +2,80 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import zeeWorkerPath from 'firefox-profiler-res/zee-worker.js';
+async function readableStreamToBuffer(
+  stream: ReadableStream<Uint8Array<ArrayBuffer>>
+): Promise<Uint8Array<ArrayBuffer>> {
+  const reader = stream.getReader();
+  const chunks: Uint8Array[] = [];
 
-const zeeCallbacks: Array<{
-  success: (data: any) => void;
-  error: (error: any) => void;
-} | null> = [];
+  try {
+    while (true) {
+      const { done, value } = await reader.read();
+      if (done) break;
+      if (value) {
+        chunks.push(value);
+      }
+    }
+  } finally {
+    reader.releaseLock();
+  }
 
-type ZeeWorkerData = {
-  callbackID: number;
-  type: 'success' | 'error';
-  data: any;
-};
+  // Calculate total length and combine chunks
+  const totalLength = chunks.reduce((sum, chunk) => sum + chunk.length, 0);
+  const result = new Uint8Array(totalLength);
+  let offset = 0;
+  for (const chunk of chunks) {
+    result.set(chunk, offset);
+    offset += chunk.length;
+  }
 
-function workerOnMessage(zeeWorker: Worker) {
-  zeeWorker.onmessage = function (msg: MessageEvent) {
-    const data = msg.data as ZeeWorkerData;
-    const callbacks = zeeCallbacks[data.callbackID];
-    if (callbacks) {
-      callbacks[data.type](data.data);
-      zeeCallbacks[data.callbackID] = null;
-    }
-  };
+  return result;
 }
 
-// Neuters data's buffer, if data is a typed array.
-export async function compress(
-  data: string | Uint8Array,
-  compressionLevel?: number
-): Promise<Uint8Array<ArrayBuffer>> {
-  if (!(typeof window === 'object' && 'Worker' in window)) {
-    // Try to fall back to Node's zlib library.
-    const zlib = await import('zlib');
-    return new Promise((resolve, reject) => {
-      zlib.gzip(data, (errorOrNull, result) => {
-        if (errorOrNull) {
-          reject(errorOrNull);
-        } else {
-          resolve(new Uint8Array(result.buffer as ArrayBuffer));
-        }
-      });
-    });
+// The Streams API doesn't support SAB, and so we need to copy from these to
+// use these API's.
```

Comment:
We're never calling these functions with shared array buffers, so it's fine to remove this function and change the input types from `Uint8Array` to `Uint8Array<ArrayBuffer>`. Unless that causes other typescript issues?

Reply:
I'm merging it as-is, we can remove this function as a follow-up.

Reply:
I tried to do that, but ran into a typescript error on the callers side. I didn't chase after it much, so it might have been an easy fix.